python/lsst/ts/pmd/component.py

```diff
@@ -180,38 +180,22 @@ async def get_slots_position(self):
             if name == "":
                 continue
             reply = await self.send_msg(str(i + 1))
-            # an empty reading returns b'', unsure what b"\r" is but was
-            # here originally
             if reply != b"\r":
                 split_reply = reply.decode().split(":")
                 position[i] = float(split_reply[-1])
+                self.log.debug(f"{position=}")
             else:
-                position[i] = math.nan
-                # Try to recover multiplexer
-                self.log.error(
-                    f"Sensor {name} is not reading. {self.read_error_count} continuous read errors reported"
-                )
-
-                if self.read_error_count == 2 and self.multiplexer_recovery():
-                    self.log.info("Multiplexer recovered")
-                    self.read_error_count = 0
-                else:
-                    self.read_error_count += 1
-                    self.log.error(
-                        "Multiplexer recovery failed "
-                        f"{self.read_error_count} times"
-                        " continuously"
-                    )
-
-        if self.read_error_count <= 3:
-            return position
-        else:
-            err_msg = "Three continuous reads of sensor ended in error."
-            self.log.exception(err_msg)
-            raise IOError(err_msg)
-
-    async def multiplexer_recovery(self):
+                # if multiplexer_recovery succeeds return position
+                # else raise IOError
+                isok = False
+                return position, isok
+        # Return positions if successful with no recovery needed.
+        isok = True
+        return position, isok
+
+    async def determine_slot_positions(self, max_resets=3):
         """Recovery of multiplexer when a sensor drops out.
+        Read slot positions with reseting the multiplexer if value fails.
```

Comment:
Please insert a blank line. Doc strings should have a brief description of no more than a line or so, a blank line, and then the rest of the info (in this case a longer description).